dallinger/experiment_server/sockets.py

```diff
@@ -87,8 +89,23 @@ def stop(self):
 
 @sockets.route('/receive_chat')
 def outbox(ws):
-    chat_backend.subscribe(ws)
+    """Sends outgoing messages to subscribed clients"""
+    channel = request.args.get('channel')
+    chat_backend.subscribe(ws, channel=channel)
 
     while not ws.closed:
         # Wait for chat backend
         gevent.sleep(0.1)
+
+
+@sockets.route('/send_chat')
+def inbox(ws):
+    """Receives incoming messages and inserts them into a Redis channel"""
+    channel = request.args.get('channel')
+
+    while not ws.closed:
+        # Sleep to prevent *constant* context-switches.
+        gevent.sleep(0.1)
```

Comment:
This probably isn't necessary here since the following statements presumably yield execution. In `outbox` it was necessary to avoid a hot loop.